aiokafka/producer.py

```diff
@@ -1,162 +1,318 @@
+            batches, one for each partition with data available to be sent.
+            A small batch size will make batching less common and may reduce
+            throughput (a batch size of zero will disable batching entirely).
+            Default: 16384
+        partitioner (callable): Callable used to determine which partition
+            each message is assigned to. Called (after key serialization):
+            partitioner(key_bytes, all_partitions, available_partitions).
+            The default partitioner implementation hashes each non-None key
+            using the same murmur2 algorithm as the java client so that
+            messages with the same key are assigned to the same partition.
+            When a key is None, the message is delivered to a random partition
+            (filtered to partitions with available leaders only, if possible).
+        max_request_size (int): The maximum size of a request. This is also
+            effectively a cap on the maximum record size. Note that the server
+            has its own cap on record size which may be different from this.
+            This setting will limit the number of record batches the producer
+            will send in a single request to avoid sending huge requests.
+            Default: 1048576.
+        metadata_max_age_ms (int): The period of time in milliseconds after
+            which we force a refresh of metadata even if we haven't seen any
+            partition leadership changes to proactively discover any new
+            brokers or partitions. Default: 300000
+        request_timeout_ms (int): Client request timeout in milliseconds.
+            Default: 30000.
+        api_version (str): specify which kafka API version to use.
+            If set to 'auto', will attempt to infer the broker version by
+            probing various APIs. Default: auto
+
+    Note:
+        Configuration parameters are described in more detail at
+        https://kafka.apache.org/090/configuration.html#producerconfigs
     """
-
-    def __init__(self, client, *,
-                 req_acks=AIOProducer.ACK_AFTER_LOCAL_WRITE,
-                 ack_timeout=1.0,
-                 codec=None,
-                 random_start=False):
-
-        self._partition_cycles = {}
-        self._random_start = random_start
-        super().__init__(client, req_acks=req_acks, ack_timeout=ack_timeout,
-                         codec=codec)
+    _PRODUCER_CLIENT_ID_SEQUENCE = 0
+
+    def __init__(self, *, loop, bootstrap_servers='localhost',
+                 client_id=None,
+                 metadata_max_age_ms=300000, request_timeout_ms=40000,
+                 api_version='auto', acks=1,
+                 key_serializer=None, value_serializer=None,
+                 compression_type=None, batch_size=16384,
+                 partitioner=DefaultPartitioner(), max_request_size=1048576):
+
+        assert acks in (0, 1, -1, 'all'), "Invalid ACKS parameter"
```

Comment:
Add a check for version. It's better for consutuctor to raise a `ValueError`, than start raising an AssertionError